### PR TITLE
docs(route53): crossaccountrole scope-down guidance

### LINKED ISSUE #28596: aws-route53: Include CrossAccountRole scope-down guidance
### Describe the issue

The `Cross Account Zone Delegation` guidance includes reference to creating a `crossAccountRole`, but provides no suggestion on how to safely scope down the role for least-privilege access. We can and should provide this guidance. 

E.g.
```
const crossAccountRole = new iam.Role(this, 'CrossAccountRole', {
  // The role name must be predictable
  roleName: 'MyDelegationRole',
  // The other account
  assumedBy: new iam.AccountPrincipal('12345678901'),
});
```
should be more like:
```
const crossAccountRole = new iam.Role(this, 'CrossAccountRole', {
      // The role name must be predictable
      roleName: 'MyDelegationRole',
      // The other account
      assumedBy: new iam.AccountPrincipal('12345678901'),
      // You can scope down this role policy to be least privileged.
      // If you want the other account to be able to manage specific records,
      // you can scope down by resource and/or normalized record names
      inlinePolicies: {
        "crossAccountPolicy": new iam.PolicyDocument({
          statements: [
            new iam.PolicyStatement({
              sid: "ListHostedZonesByName",
              effect: iam.Effect.ALLOW,
              actions: ["route53:ListHostedZonesByName"],
              resources: ["*"]
            }),
            new iam.PolicyStatement({
              sid: "GetHostedZoneAndChangeResourceRecordSet",
              effect: iam.Effect.ALLOW,
              actions: ["route53:GetHostedZone", "route53:ChangeResourceRecordSet"],
              // This example assumes the RecordSet subdomain.somexample.com 
              // is contained in the HostedZone
              resources: ["arn:aws:route53:::hostedzone/HZID00000000000000000"],
              conditions: {
                "ForAllValues:StringLike": {
                  "route53:ChangeResourceRecordSetsNormalizedRecordNames": [
                  "subdomain.someexample.com"
                ]

                }
              }
            })
    });
```

### Links

https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_route53-readme.html#cross-account-zone-delegation

### PR DESCRIPTION
**Pull request by miiiak**
Reference [issue 28596](https://github.com/aws/aws-cdk/issues/28596)

The motivation is to help CDK builders understand how to take advantage of IAM scope-down capabilities to ensure least-privilege cross-account role access related to cross account zone delegation.

The Cross Account Zone Delegation guidance currently includes reference to creating a crossAccountRole, but provides no suggestion on how to safely scope down the role for least-privilege access. We can and should provide this guidance.

E.g.
```
const crossAccountRole = new iam.Role(this, 'CrossAccountRole', {
  // The role name must be predictable
  roleName: 'MyDelegationRole',
  // The other account
  assumedBy: new iam.AccountPrincipal('12345678901'),
});
```
should be more like:
```
const crossAccountRole = new iam.Role(this, 'CrossAccountRole', {
      // The role name must be predictable
      roleName: 'MyDelegationRole',
      // The other account
      assumedBy: new iam.AccountPrincipal('12345678901'),
      // You can scope down this role policy to be least privileged.
      // If you want the other account to be able to manage specific records,
      // you can scope down by resource and/or normalized record names
      inlinePolicies: {
        "crossAccountPolicy": new iam.PolicyDocument({
          statements: [
            new iam.PolicyStatement({
              sid: "ListHostedZonesByName",
              effect: iam.Effect.ALLOW,
              actions: ["route53:ListHostedZonesByName"],
              resources: ["*"]
            }),
            new iam.PolicyStatement({
              sid: "GetHostedZoneAndChangeResourceRecordSet",
              effect: iam.Effect.ALLOW,
              actions: ["route53:GetHostedZone", "route53:ChangeResourceRecordSet"],
              // This example assumes the RecordSet subdomain.somexample.com 
              // is contained in the HostedZone
              resources: ["arn:aws:route53:::hostedzone/HZID00000000000000000"],
              conditions: {
                "ForAllValues:StringLike": {
                  "route53:ChangeResourceRecordSetsNormalizedRecordNames": [
                  "subdomain.someexample.com"
                ]

                }
              }
            })
    });
```
Closes #28596.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by miiiak** [comment]
Clarification Request - Are test file / integration tests required for README updates? If yes, should I assume the necessary test is to prove that the updated snippets synthesize and build correctly, or what is the expectation?

File: packages/aws-cdk-lib/aws-route53/README.md [hunk 1]
```diff
@@ -192,11 +192,39 @@ const parentZone = new route53.PublicHostedZone(this, 'HostedZone', {
...
+      // If you want the other account to be able to manage specific records,
+      // you can scope down by resource and/or normalized record names
+      inlinePolicies: {
+        "crossAccountPolicy": new iam.PolicyDocument({
+          statements: [
+            new iam.PolicyStatement({
+              sid: "ListHostedZonesByName",
+              effect: iam.Effect.ALLOW,
+              actions: ["route53:ListHostedZonesByName"],
+              resources: ["*"]
```
Comment (kaizencc) [inline packages/aws-cdk-lib/aws-route53/README.md:209]: nit: i want a comma here and in all other places that should have punctuation

File: packages/aws-cdk-lib/aws-route53/README.md [hunk 2]
```diff
@@ -192,11 +192,39 @@ const parentZone = new route53.PublicHostedZone(this, 'HostedZone', {
   zoneName: 'someexample.com',
 });
 const crossAccountRole = new iam.Role(this, 'CrossAccountRole', {
-  // The role name must be predictable
-  roleName: 'MyDelegationRole',
-  // The other account
-  assumedBy: new iam.AccountPrincipal('12345678901'),
-});
+      // The role name must be predictable
```
Comment (kaizencc) [inline packages/aws-cdk-lib/aws-route53/README.md:195]: the formatting here is inconsistent with the rest of the doc in terms of tabs

**Review by kaizencc — DISMISSED** [review]
I'm happy to accept this, and thanks for improving our docs. However, the build has to succeed, and we run our readme examples to make sure they compile (and yours does not). once yours does, i'll approve.

to check if your example compiles, you can run `yarn rosetta` after building aws-cdk-lib

File: packages/aws-cdk-lib/aws-route53/README.md (hunk 1 above)
Comment (miiiak) [reply packages/aws-cdk-lib/aws-route53/README.md:209]: Ack, should be resolved.

File: packages/aws-cdk-lib/aws-route53/README.md (hunk 2 above)
Comment (miiiak) [reply packages/aws-cdk-lib/aws-route53/README.md:195]: Not sure I follow, I see other comments in the doc follow the tab alignment of the line of code below it that it speaks to. Can you clarify?

Comment (kaizencc) [reply packages/aws-cdk-lib/aws-route53/README.md:195]: Screenshot 2024-01-09 at 3 26 23 PM


here is your rich diff. the tab alignment is off

**Review by kaizencc — APPROVED** [review]
Thanks @miiiak!

**Pull request merged**